### ISSUE TITLE
Package request: Gnome shell

### ISSUE THREAD

**Issue by suhan-paradkar** [termux/x11-packages#314]
**Package description**

GNOME Shell provides core user interface functions for the GNOME 3 desktop,
like switching to windows and launching applications. GNOME Shell takes
advantage of the capabilities of modern graphics hardware and introduces
innovative user interface concepts to provide a visually attractive and
easy to use experience.

**Link to home page and sources**
1. Home page:https://wiki.gnome.org/Projects/GnomeShell
2. Source code:https://gitlab.gnome.org/GNOME/gnome-shell

I tried compiling it... It requires a lot of absent dependencies... I will be creating pull requests for some of the depenencies...

**Comment by suhan-paradkar** [comment]
What do you think @WMCB-Tech ?

**Comment by WMCB-Tech** [comment]
GNOME 3 is very nice, although i don't think it will ever be ported as 

* It requires systemd?

* Most of the time, dependencies will be absent and may have issues with cross compilation

* Performance loss will occur as it's heavy, and may require OpenGL (atleast GLX extension)

Although if you think that you can compile it successfully, then give it a go

You could try compiling MATE instead as it's lighter

**Comment by suhan-paradkar** [comment]
Well... Systemd integration can be disabled using  while configuring using ```-Dsystemd=false```

**Comment by WMCB-Tech** [comment]
I see, but as far as i know, GNOME also depends on `gobject-introspection` which it has problems with Cross Compilation

Also most likely it may get errors regarding with **Oh no** failed whale screen, even after things got compiled successfully

**Comment by suhan-paradkar** [comment]
> I see, but as far as i know, GNOME also depends on `gobject-introspection` which it has problems with cross compilation afaik (idk actually, please correct me)
> 
> Also in some cases it may get errors regarding with **Oh no** failed whale screen when running GNOME session even after things got compiled successfully

Yeah.. you are right.....that may happen.. and gobject introspection cannot be compiled even natively on device... So... Yeah

**Comment by suhan-paradkar** [comment]
> GNOME 3 is very nice, although i don't think it will ever be ported as
> 
>     * It requires systemd?
> 
>     * Most of the time, dependencies will be absent and may have issues with cross compilation
> 
>     * Performance loss will occur as it's heavy, and may require OpenGL (atleast GLX extension)
> 
> 
> Although if you think that you can compile it successfully, then give it a go
> 
> You could try compiling MATE instead as it's lighter but still not sure if that would be possible

@WMCB-Tech I was able to compile marco window manager (default window manager of MATE) natively...

**Comment by LeapofAzzam-backup** [comment]
Well, it was very heavy and not suitable for Termux

**Comment by LeapofAzzam-backup** [comment]
But we should be able to port GNOME Flashback

**Comment by Yisus7u7** [comment]
> Pero deberíamos poder portar GNOME Flashback

I do not think it will be ported, apart very soon it will be added matte (a fork of gnome 2), at the moment in termux is available xfce4 and lxqt

**Comment by suhan-paradkar** [comment]
> I see, but as far as i know, GNOME also depends on `gobject-introspection` which it has problems with cross compilation afaik (idk actually, please correct me)

Well as gobject-introspection is now compiled... We can proceed ahead now....
gnome-shell now requires evolution-data-server which requires nspr and nss... Both of which are currently absent....

**Comment by layaachmad** [comment]
GNOME aren't possible on Termux

**Comment by Yisus7u7** [comment]
> GNOME no es posible en Termux

Nothing in this life is impossible

**Comment by Yisus7u7** [comment]
@suhan-paradkar How is progress going? Hahaha

**Comment by suhan-paradkar** [comment]
Well.. I require help porting mozjs tho... (A dependency of `gjs`)

**Comment by Yisus7u7** [comment]
@suhan-paradkar @wmcbtech30

And if we use Cinnamon?, I see what is lighter in dependencies.
